code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -225,35 +290,40 @@
 		worklocation()
 		playsound(loc, 'sound/machines/twobeep.ogg', 50, 1)	//Plays a beep
 		visible_message("Shuttle Locator mode actived.")			//Lets the mob holding it know that the mode has changed
+		active = 1
 		return		//Get outta here
 	scandisk()
-	if(!the_disk)
-		icon_state = icon_null
+	point_at(the_disk)
+
+/obj/item/pinpointer/nukeop/workbomb()
+	if(active != 2)
 		return
-	dir = get_dir(src, the_disk)
-	switch(get_dist(src, the_disk))
-		if(0)
-			icon_state = icon_direct
-		if(1 to 8)
-			icon_state = icon_close
-		if(9 to 16)
-			icon_state = icon_medium
-		if(16 to INFINITY)
-			icon_state = icon_far
-
-	spawn(5) .()
+	if(mode)		//Check in case the mode changes while operating
```

Comment:
Should not happen anymore due to the elimination of spawn

Reply:
dealt with